### ISSUE TITLE
Efficientnet-edgetpu-S shape incorrect with custom dataset

### ISSUE THREAD

**Issue by BernardinD** [tensorflow/tpu#763]
I have been trying to use the efficientnet-edgetpu-S model and I've been having trouble with running the inference on the exported saved_model and getting the same results as when running the ckpts through the model Driver.

I believe the issue is related to the fact that when the graph is exported to a saved_model the shape of the output tensor does not equal the number of classes specified for training's _--num_classes_ parameter, which is equal to 50. But instead its still equal to the original default number of classes (1001).

After running _**saved_model_cli show --all --dir {pb_dir}**_, this is the response for the shape of the input and output:
`MetaGraphDef with tag-set: 'serve' contains the following SignatureDefs:

signature_def['serving_default']:
  The given SavedModel SignatureDef contains the following input(s):
    inputs['input'] tensor_info:
        dtype: DT_FLOAT
        shape: (1, 224, 224, 3)
        name: images:0
  The given SavedModel SignatureDef contains the following output(s):
    outputs['output'] tensor_info:
        dtype: DT_FLOAT
        shape: (1, 1001)
        name: Softmax:0
  Method name is: tensorflow/serving/predict`

The correct shape is in places like the _model_params.txt_ file from training and presumed in other places as well, but the conflicting shapes show up when attempting to run the saved_model or a frozen graph of the saved_model.

Any help is appreciated. The overall goal is to get the efficientnet-edgetpu-S graph to run from a frozen graph after being trained a custom dataset and a specific number of classes

**Comment by BernardinD** [comment]
After looking into it, it seems the problem was a hard-coded number of classes in the builder file for the model. It doesn't seem to adjust the _override_params_ variable like the training script does.
